cmd/protoc-gen-go-tableau-checker/hub.go

```diff
@@ -29,89 +29,104 @@ const staticHubContent = `
 	"github.com/tableauio/tableau/load"
 	"github.com/tableauio/tableau/log"
 	"github.com/tableauio/tableau/proto/tableaupb"
-	"golang.org/x/sync/errgroup"
+	"github.com/tableauio/tableau/xerrors"
 	"google.golang.org/protobuf/proto"
 	"google.golang.org/protobuf/reflect/protoreflect"
 	"google.golang.org/protobuf/reflect/protoregistry"
 )
 
-var registrarSingleton *tableau.Registrar
+type Checker interface {
+	tableau.Messager
+	Messager() tableau.Messager
+	Check(hub *tableau.Hub) error
+	CheckCompatibility(hub, newHub *tableau.Hub) error
+}
+
+type CheckerMap = map[string]Checker
+type CheckerGenerator = func() Checker
+type Registrar struct {
+	Generators map[string]CheckerGenerator
+}
+
+func NewRegistrar() *Registrar {
+	return &Registrar{
+		Generators: map[string]CheckerGenerator{},
+	}
+}
+
+func (r *Registrar) Register(gen CheckerGenerator) {
+	if _, ok := r.Generators[gen().Name()]; ok {
+		panic("register duplicate checker: " + gen().Name())
+	}
+	r.Generators[gen().Name()] = gen
+}
+
+var registrarSingleton *Registrar
 var once sync.Once
 
-func getRegistrar() *tableau.Registrar {
+func getRegistrar() *Registrar {
 	once.Do(func() {
-		registrarSingleton = tableau.NewRegistrar()
+		registrarSingleton = NewRegistrar()
 	})
 	return registrarSingleton
 }
 
+func Register(gen CheckerGenerator) {
+	getRegistrar().Register(gen)
+}
+
 type Hub struct {
 	*tableau.Hub
-	filteredCheckerMap tableau.MessagerMap
+	filteredCheckerMap CheckerMap
```

Comment:
Remove the definition: `type CheckerMap = map[string]Checker`, and just use:

`filteredCheckerMap CheckerMap` -> `checkers map[string]Checker`